src/models/EventMapper.php

```diff
@@ -100,6 +100,32 @@ public function getEventById($event_id, $verbose = false, $activeEventsOnly = tr
     }
 
     /**
+     * Find an event from its track id
+     *
+     * Optionally return the non-transformed data.
+     *
+     * @param int  $track_id
+     * @param bool $verbose
+     * @param bool $activeEventsOnly
+     * @param bool $transform
+     *
+     * @return array the event detail
+     */
+    public function getEventByTrackId($track_id, $verbose = false, $activeEventsOnly = true, $transform = true)
+    {
+        $results = $this->getEvents(1, 0, array("track_id" => $track_id, 'active' => $activeEventsOnly));
+        if (!$results) {
+            return false;
```

Comment:
Wouldn't it be easier to throw an exception here? Now we always have to check whether ```getEventByTrackId``` returns ```false``` to take appropriate measures. With an Exception that would be wrapping the call into a ```try-catch```-block or just passing the exception on…